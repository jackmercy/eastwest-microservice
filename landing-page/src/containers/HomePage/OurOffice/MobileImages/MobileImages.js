import { memo, useMemo } from 'react';
import PropTypes from 'prop-types';

import { Carousel } from '../../../../components';

import useStyles from './mobileImages.styles';

const MobileImages = ({ images }) => {
  const classes = useStyles();
  const settings = useMemo(
    () => ({
      dots: true,
      nextArrow: null,
      prevArrow: null,
    }),
    [],
  );

  return (
    <Carousel {...settings} className={classes.carousel}>
      {images.map((item) => (
        <div key={`our-office-image-${item.key}`}>
          <img
            alt="Eastwest office"
            src={item.src}
            srcSet={item.srcSet}
            className={classes.image}
          />
        </div>
      ))}
    </Carousel>
  );
};
MobileImages.propTypes = {
  images: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string.isRequired,
      src: PropTypes.string.isRequired,
      srcSet: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default memo(MobileImages);
