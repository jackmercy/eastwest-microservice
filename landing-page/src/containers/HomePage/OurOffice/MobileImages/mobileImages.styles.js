import { createUseStyles } from 'react-jss';

import { rem } from '../../../../utils/jss';

export default createUseStyles({
  carousel: {
    '& .slick-slide > div': { width: '100%' },
  },
  image: { display: 'block', maxHeight: rem(230) },
});
