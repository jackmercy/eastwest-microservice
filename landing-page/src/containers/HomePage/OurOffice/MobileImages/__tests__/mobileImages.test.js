import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import MobileImages from '../MobileImages';

describe('Testing HomePage > OurOffice > MobileImages', () => {
  const props = {
    images: [
      {
        key: 'cover-1',
        src: '/static/images/homepage/office-01.png',
        srcSet: [
          '/static/images/homepage/office-01.png 500w',
          '/static/images/homepage/office-01@2x.png 1500w',
          '/static/images/homepage/office-01@3x.png 2000w',
        ].join(', '),
      },
      {
        key: 'cover-2',
        src: '/static/images/homepage/office-02.png',
        srcSet: [
          '/static/images/homepage/office-02.png 500w',
          '/static/images/homepage/office-02@2x.png 1500w',
          '/static/images/homepage/office-02@3x.png 2000w',
        ].join(', '),
      },
    ],
  };

  it('should render match snapshot', () => {
    const wrapper = shallow(<MobileImages {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
