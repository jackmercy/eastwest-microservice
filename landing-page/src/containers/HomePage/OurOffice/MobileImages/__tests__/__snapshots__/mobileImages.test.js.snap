// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomePage > OurOffice > MobileImages should render match snapshot 1`] = `
<Memo(Carousel)
  className="carousel-0-2-1"
  dots={true}
  nextArrow={null}
  prevArrow={null}
>
  <div
    key="our-office-image-cover-1"
  >
    <img
      alt="Eastwest office"
      className="image-0-2-2"
      src="/static/images/homepage/office-01.png"
      srcSet="/static/images/homepage/office-01.png 500w, /static/images/homepage/office-01@2x.png 1500w, /static/images/homepage/office-01@3x.png 2000w"
    />
  </div>
  <div
    key="our-office-image-cover-2"
  >
    <img
      alt="Eastwest office"
      className="image-0-2-2"
      src="/static/images/homepage/office-02.png"
      srcSet="/static/images/homepage/office-02.png 500w, /static/images/homepage/office-02@2x.png 1500w, /static/images/homepage/office-02@3x.png 2000w"
    />
  </div>
</Memo(Carousel)>
`;
